rust/connlib/libs/tunnel/src/peer.rs

```diff
@@ -83,7 +104,17 @@ impl Peer {
             .is_some_and(|expires_at| expires_at <= Utc::now())
     }
 
-    pub(crate) fn is_allowed(&self, addr: impl Into<IpAddr>) -> bool {
+    pub(crate) fn is_allowed(&self, addr: IpAddr) -> bool {
         self.allowed_ips.longest_match(addr).is_some()
     }
+
+    pub(crate) fn update_translated_resource_address(&self, addr: IpAddr) {
+        if !self
+            .translated_resource_address
+            .read()
+            .is_some_and(|stored| stored == addr)
```

Comment:
It may be simpler to just `== Some(addr)` with the result of `read` instead of using `is_some_and` here

Reply:
I think it looks better like this otherwise the condition looks like:
```rs
        if !(*self.translated_resource_address.read() == Some(addr)) {
```

which I think it's harder to parse at a simple glance

Reply:
Ah, I didn't notice the `!`... surely it's clearer if you use `!=`?

Reply:
It's clearer but I still like not explicitly de-referencing the `RWLockReadGuard`.

Reply:
Fair enough